### DISCUSSION TITLE
ESMValTool Workshop (in-person/hybrid) dates are fixed!

### DISCUSSION THREAD

**Discussion by hb326** [ESMValGroup/Community#21] [Meetings]
Hey all,
the dates for the next ESMValTool workshop are fixed now: **June 13-15 (Monday - Wednesday).**
Looking forward to seeing many of you hopefully in person!

**Comment by hb326** [comment]
Hey all again @ESMValGroup/esmvaltool-developmentteam,
we are starting to collect possible session items and discussion points for the upcoming ESMValTool workshop. So please, if you have anything that you think should be discussed in a dedicated session at the workshop, comment here and we will collect all the ideas. 
Thanks!!

**Comment by rswamina** [comment]
A discussion on tutorials and the kind of tutorials (project specific, regularly-running, topic-specific) we should organize and provide will be good.

**Reply by hb326** [reply]
Yes, we should do that. 
I think we also thought about giving an introduction about the User Engagement Team, and what we do and can provide?

**Comment by remi-kazeroni** [comment]
I'd like to propose a session on obs data. There is growing needs for obs data for several projects and on shared analysis platforms. It would be good to discuss a few related topics and reach consensus on how to move forward with these data. Unfortunately, I will be able to attend the workshop myself but I'm happy to provide an agenda for this session if someone is willing to chair it at the workshop. Possible questions to discuss with links to ongoing issues and discussions:

- **Short demo on the new cmorizer interface** and added features (see https://github.com/ESMValGroup/ESMValTool/pull/1657 and https://github.com/ESMValGroup/ESMValTool/discussions/2560)
- Would it be needed to **revise the current tier classification?** There is increasing interest in accessing our shared pools of obs data (DKRZ, CEDA-Jasmin). It is always unclear if we should give anyone access to our pools of cmorized Tier3 datasets (see https://github.com/ESMValGroup/ESMValTool/issues/1072 and https://github.com/ESMValGroup/ESMValTool/pull/2385). A possible way forward would be to ask users to type `esmvaltool data info <dataset>`, conform to the instructions displayed. But that would need to be repeated for every tier3 dataset of interest.
- Do we want to **manage the distribution of data between main centres using ESMValTool** and how? To avoid data replications on given clusters, people in charge could set up local replica of the most up-to-date data collection (DKRZ) for the datasets of interests (see https://github.com/ESMValGroup/ESMValTool/issues/2630). At the moment there is only partial synchronization of data from DKRZ to Jasmin (rsync by myself).
- How could we **keep our cmorizers up-to-date when raw data are updated upstream**: added timerange, newer versions, etc. see https://github.com/ESMValGroup/ESMValTool/issues/2591. Would it be worth the effort to setup a mechanism to automatically check if raw data have been updated upstream? Updating cmorizers would mean, in most cases, to edit the code (yml and python files) because of the hard-coded version numbers, filenames, paths, etc. and checking new outputs. How could we keep these updates lightweight if that is desirable?

**Comment by hb326** [comment]
News about the ported webpage to Github where people beside DLR employees can add things.

**Comment by alistairsellar** [comment]
An update + discussion on progress towards a new policy on backward compatibility.

**Comment by schlunma** [comment]
**Discussion on the long-term strategy of ESMValTool.**

This can include, but is not limited to,

- the implementation of a set of basic diagnostics that provide the most popular plot types. These can be used for quick evaluations without the need to write dedicated diagnostics and will provide a much easier starting point for new users.
- a prioritization system of all diagnostics; diagnostics with low priority will not be tested for every single release.

Two main aims of these suggestions are to make ESMValTool more beginner-friendly and to reduce the workload for the release managers.

**Reply by alistairsellar** [reply]
Good idea. Another long-term strategy topic is
 - What is needed to get ESMValTool exascale-ready? (E.g. to cope with higher resolution models, more "funny" grids, and also any implications of changes in the platforms used for modelling.)

**Reply by bouweandela** [reply]
We might want to start with getting the tool 'terascale'-ready, before even thinking about exascale..

**Reply by bouweandela** [reply]
But maybe that's just me being cynical, it's always good to look ahead :+1:

**Reply by alistairsellar** [reply]
Agreed, but as you say it's important to look ahead. An agreed long-term strategy for future capability will help to motivate project bids, and identify collaboration needs & opportunities. Several countries already have exascale modelling programmes, and these should include plans to ensure that the models can be evaluated. If ESMValTool doesn't look like it will be up to the challenge then these programmes will re-invent the wheel themselves. (We have already seen at least one such example of a programme proposing to do this, though admittedly that was driven by them not being aware of current ESMValTool capability rather than lack of confidence in future exascale suitability.)

**Comment by hb326** [comment]
Hey all @ESMValGroup/esmvaltool-developmentteam,
we are confirming available rooms for our meeting at DLR right now, and would need a pretty exact head count of people who would want to attend the meeting in person from the the list of people who have signed up for the workshop, and everybody else interested who had not signed up.
Could you please let us know (either by commenting here or sending me an email) by **latest Friday May 20**?
Thanks!

**Comment by hb326** [comment]
Hey all @ESMValGroup/esmvaltool-developmentteam,
please find below a preliminary workshop agenda. 
Please comment on the agenda **by June 2** and let us know:

1. If you would like to have more (actually _any_) coding time at the workshop? If yes, what would you like to work on and what should be taken off the agenda to make time for the coding?
2. If you are volunteering for chairing a session. We still need some spots filled.

Thanks!


### **Workshop agenda (updated 9 June 2022)**

**13 June 2022 - Day 1 (Monday)**
10.15 am – 10.20 am 	Introduction, welcome and workshop agenda (Birgit)
10.20 am – 10.30 am 	Update and current status of the consortium agreement and contributing license agreement (CLA) (Alistair)
10.30 am – 10.45 am	Report from Technical Lead Development Team (V)
10.45 am – 11.00 am	Report from Science Lead Development Team (Axel)
11.00 am – 11.15 am	Break
11.15 am – 12.15 pm 	User Engagement Team

- Introduction – Tasks and responsibilities of the User Engagement Team (Lisa)
- Strategy and types of tutorials (Ranjini)

12.15 pm – 01.30 pm Break
01.30 pm – 03.30 pm	Scientific long-term strategy

- Introduction and main aims of diagnostics (Axel)
- Porting of diagnostics from NCL to Python (Axel)
- Dealing with recipe maintainers no longer active (Lee, Bouwe)
- Prioritizing diagnostics for testing (releases) (Manuel)
- Addition of set of generic diagnostics (Manuel, Saskia)

03.30 pm – 03.45 pm	Break
03.45 pm – 04.00 pm	Summary and wrap up of first day (Birgit)


**14 June 2022 - Day 2 (Tuesday)**
10.15 am – 10.20 am 	Introduction day 2 and workshop agenda (Birgit)
10.20 am – 12.00 pm 	Observations

- Short demo on the new cmorizer interface (Manuel)
- Revision of current tier classification (Axel, Birgit)
- Distribution of data between computing centers (Klaus)
- Automatic updates and versioning of observational datasets (Klaus)

12.15 pm – 01.30 pm Break
01.30 pm – 02.30 pm	Strategy for handling and communicating implementation of non-backward compatible changes with the aim of minimizing unnoticed breaking of recipes new changes (Alistair)
02.30 pm – 02.45 pm	Break
02.45 pm – 03.45 pm	Testing strategy (including automatic comparison of results (Birgit, Bouwe)
03.45 pm – 04.00 pm	Break
04.00 pm – 05.00 pm	Scientific long-term strategy (continued from Monday) (Axel, Manuel, Klaus, Saskia)
05.00 pm – 05.15 pm	Summary and wrap up of second day (Birgit)

06.30 pm			Workshop dinner


**15 June 2022 - Day 3 (Wednesday)**
10.15 am – 10.20 am 	Introduction day 3 and workshop agenda (Birgit)
10.20 am – 12.00 pm 	Technical strategy for ESMValTool v3.0 including e.g., (V, Bouwe, Klaus)

- Higher resolution models
- Handling of complex grids including CORDEX datasets
- Parallelization, memory management, cloud computing

12.15 pm – 01.30 pm Break
01.30 pm – 03.00 pm	General discussion

- New ESMValTool website (Axel)
- Handling of inactive developers (Alistair)
- Optimization of the tool’s performance (Bouwe)
- Future funding and new road map (Birgit)

03.00 pm – 03.15 pm	Break
03.15 pm – 04.00 pm	Wrap up (Alistair, Birgit)

**Comment by sloosvel** [comment]
> 10.20 am – 12.00 pm Technical strategy for ESMValTool v3.0 (?)
>...
>Handling of complex grids

Is there any chance this topic could include a discussion on Cordex datasets? There is several work on them that has been started, but has been left abandoned (fyi @pepcos)

**Reply by axel-lauer** [reply]
Good point, thanks for this idea! I just added CORDEX to "handling of complex grids" in this session.

**Reply by zklaus** [reply]
For additional context: CORDEX grids are not complex like ICON or FESOM grids, but rather at the level of the tripolar ocean grids and as such have a similar level of support already. The problem with CORDEX grids is usually that there are metadata difficulties, such as incorrect or missing coordinates, minuscule rounding differences in coordinates even within the same model between historical and scenario runs etc.

**Comment by remi-kazeroni** [comment]
> 01.30 pm – 03.00 pm General discussion
> 
>   New ESMValTool website (Axel)
>   Handling of inactive developers (Alistair?)
>   Optimization of the tool’s performance (Bouwe?)
>   Future funding and new road map (Birgit)

As part of the session on future funding, it would be good to talk about the opportunity to join the future ENES-RI (see [here](https://portal.enes.org/community/announcements/events/enes-research-infrastructure-scoping-workshop) the announcement for the ENES-RI introductory meeting of last year), how we could contribute and what the benefits for ESMValTool would be. The participation to this structure would have to be decided by each institute individually but it may help to know if (some) ESMValTool Core institutes are planning to join. ENES-RI is designed to be the self-sustained successor of [IS-ENES3](https://is.enes.org/) (2019-2023) whose funding ends in March 2023 and will (most likely) not be extended for a fourth phase.

**Comment by axel-lauer** [comment]
The workshop agenda (https://github.com/ESMValGroup/Community/discussions/21#discussioncomment-2812887) is now also available as pdf (including a link to the video meeting via webex). Thank you to everyone for helping us with organizing and chairing the sessions! Looking forward to seeing you next week!

[Agenda_ESMValTool_WS_2022-06.pdf](https://github.com/ESMValGroup/Community/files/8869651/Agenda_ESMValTool_WS_2022-06.pdf)
